Recreate blocks using only wp core - remove all css + Min WP Version Bump to 5.8

godleman opened the pull request:
<!--
Filling out this template is required.  Any PR that does not include enough information to be reviewed may be closed at a maintainers' discretion.  All new code requires documentation and tests to ensure against regressions.
-->

### Description of the Change

Aims of these changes:
WP patterns cannot contain CSS. We need to refactor using only wp core settings in order to prepare to add patterns to WP pattern library.

Actions from this change:

- Recreating the blocks using current WP core block options only. 
- Replacing old blocks with wp-core generated source code. 
- Removing all block css in favour of inlined styling from wp core settings.
- Replacing all images with public urls from openverse
- Also bumping the WP minimum required version from `5.7` to `5.8` because, after this PR, the block patterns will be broken in 5.7 or below.

Scope outlined by @jeffpaul 
See @jeffpaul for documentation if needed.

<!--
We must be able to understand the design of your change from this description.  The maintainer reviewing this PR may not have worked with this code recently, so please provide as much detail as possible.

Where possible, please also include:
- verification steps to ensure your change has the desired effects and has not introduced any regressions
- any benefits that will be realized
- any alternative implementations or possible drawbacks that you considered
- screenshots or screencasts
-->

<!-- Enter any applicable Issue number(s) here that will be closed/resolved by this PR. -->
Closes #

### How to test the Change
<!-- Please provide steps on how to test or validate that the change in this PR works as described. -->
Create the patterns in the plugin in wp and observe the output in the editor and FE. 

### Changelog Entry
<!--
Please include a summary for this PR, noting whether this is something being Added / Changed / Deprecated / Removed / Fixed / or Security related.  You can replace the sample entries after this comment block with the single changelog entry line for this PR. -->
> Added - WP Core only block code
> Changed - Existing block code and css
> Deprecated - n/a
> Removed - Existing block code and css
> Fixed - n/a
> Security - n/a


### Notes / Issues in recreating design using only wp core blocks settings
**Global content width**
No option change, so the narrow 650px content width restricts the design. For instance font sizes need to be decrease otherwise they'll look oversized against the narrow content width

**Group blocks** 
You can add borders but you can't limit the border stroke to only the bottom of the element - it's always a full element border. You can only control the border width and style. 

**Text elements** 
Margin - you can add positive margin values but you cannot reset core text element margins or apply negative margin values to compensate. 

**Tables**
Stripes - you can add stipes but you can't control the stripe background color value - its always grey 

**Tabs**
Heading - There is a heading expectation when you create a Tabs Block. But this heading does not have the usual Heading element controls like font-size, appearence etc. So it is best to create a core Heading element above the tabs block. But the Tabs block blank heading still renders and adds unwanted margin top and bottom. 

**Tabs Item** 
There are not styling controls on Tab Items - like modifying or removing the default borders or modifying the Tab heading padding to match a design 

**Columns** 
Style - no option to add border or box-shadow - so you can't replicate box Column box shadows from design

**Lists** - no option to replace standard HTML list item bullets with images/icons - so better to just use a 'Row' with an image and a paragraph block. 

**Row** - You can justify items but you can't align items - so you can't have a row with top aligned content, its always center aligned

### Credits
<!-- Please list any and all contributors on this PR so that they can be added to this projects CREDITS.md file. -->
Props @username, @username2, ...


### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you are unsure about any of these, please ask for clarification.  We are here to help! -->
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

godleman committed:
Recreate blocks using only wp core - remove all css

faisal-alvi reviewed (changes requested):
@cldhmmr thank you for working on this. I have tested the PR. Please check the following points that I observed while testing:

1. The cover block pattern seems to be broken:

![image](https://user-images.githubusercontent.com/25176325/196993897-f0cbede4-76b0-43ad-b6ca-62215ace8cde.png)

2. All blocks have the same title "2.5M",

![image](https://user-images.githubusercontent.com/25176325/196995045-c6ef2b35-2b64-4b65-84ef-0533dccf0c48.png)

it should be different for each block:

![image](https://user-images.githubusercontent.com/25176325/196995166-094b4565-6246-40fa-9478-8b7b0f90c7af.png)

3. The active tab item font size is tiny, not sure if this is caused by the current implementation, but it would be great if it's fixed. Tested with the Storefront theme.

![image](https://user-images.githubusercontent.com/25176325/196997472-a1d4c0c0-2674-4ba5-859f-a4528cc89ba3.png)

4. Our Packages section seems broken, looks like the bullet images are 404 not found.

![image](https://user-images.githubusercontent.com/25176325/196998127-f91dff5b-0f21-4b8d-8551-8e09e2553e53.png)

5. Also the Media Kit page is not 100% responsive now, we can see the large fonts in the Tabs block and the CTA inline button. Should we keep it as it is? cc @jeffpaul 

![image](https://user-images.githubusercontent.com/25176325/196999029-877729a0-2608-4ab5-8e8b-ddf0aae07617.png)

![image](https://user-images.githubusercontent.com/25176325/196998969-5c59caea-a138-428d-a043-b2afa7631a72.png)

6. The font sizes are different in the Audience Profile pattern:

![image](https://user-images.githubusercontent.com/25176325/197000430-89f0c91a-a150-443b-ab3a-40b3c4c16b9d.png)

7. The cypress test is [failing](https://github.com/10up/publisher-media-kit/actions/runs/3290571184/jobs/5423590603) for WP#5.7. Seems like the test can not find the `core/heading` element.

![image](https://user-images.githubusercontent.com/25176325/196999570-172229ee-f693-4e63-a9a5-df974572ffaf.png)

godleman commented:
Thanks @faisal-alvi 
This wasn't happening locally, I'll look in to these and update asap

godleman committed:
Updates as per CR comments

godleman commented:
@faisal-alvi cc @jeffpaul  I've made some changes, please review my responses below. I think its best to recreate the blocks fresh to see the changes :

**The cover block pattern seems to be broken:**
I've changed the image url for this and the block seems ok. 
![image](https://user-images.githubusercontent.com/55838845/197181349-7b23ebf1-1bc0-4e84-bdf0-6ce778e402a1.png)

**All blocks have the same title "2.5M":**
Updated to variations thanks 

**The active tab item font size is tiny, not sure if this is caused by the current implementation, but it would be great if it's fixed. Tested with the Storefront theme.**
I've updated this in the Tab block editor styles. 
![image](https://user-images.githubusercontent.com/55838845/197186791-c1694999-b430-4ce8-8c01-02cc36855b3a.png)


**Our Packages section seems broken, looks like the bullet images are 404 not found.**
The images are now updated to public urls - they were referencing local, so this should be fixed 
![image](https://user-images.githubusercontent.com/55838845/197186755-462a1b41-8dbc-4ddd-914d-c09561e52944.png)


**Also the Media Kit page is not 100% responsive now, we can see the large fonts in the Tabs block and the CTA inline button. Should we keep it as it is? cc @jeffpaul**

I haven't been able to replicate an issue with the table itself - the table show a horizontal overflow scrollbar on mobile and the user can slide that. Could you verify please?
![image](https://user-images.githubusercontent.com/55838845/197186912-04f98112-d228-48df-b4c0-07330af530cd.png)

I agree with the Still Have Questions CTA block, there is currently not a setting in wordpress core to control the layout on mobile, so we are stuck with the row alignment in this case on mobile.
![image](https://user-images.githubusercontent.com/55838845/197187371-15bb9199-c329-411b-9f37-7e235d1b432e.png)

**The font sizes are different in the Audience Profile pattern:**
This seemed to be a inconsistency in the output from editor - I think this is fixed now. I have also changed the images to be more like the original design using wiki commons images. I am a little bit hesitant about the images being outdated gender archetypes and would suggest we change to something non-gendered like animals or lego, or something else. Thoughts on this appreciated. 
![image](https://user-images.githubusercontent.com/55838845/197188017-a656afc0-e288-4364-833a-75714f3903b1.png)

**The cypress test is [failing](https://github.com/10up/publisher-media-kit/actions/runs/3290571184/jobs/5423590603) for WP#5.7. Seems like the test can not find the core/heading element.**

I don't understand why this is being thrown or how to fix this? I'm not sure if related to the updates or not. Please let me know if you have any ideas?

**Important note about images**
A lot of time is spent around images and images changes. 
Issues are :
Wordpress want us to use OpenVerse 
OpenVerse images are often sourced from personal Flickr accounts which we wouldn't really want to embed in our plugin
OpenVerse in general has a weak search facility and it can take a long time to find appropriate images
When logged into wordpress.org you can create a new pattern directly and the openverse image search is integrated, so searching here and using url from there seems like a good idea. But, there are 2 problems. a) the search facilty is terrible, even worse than the main openverse site, and b) the editor often does not like the image urls and they don't display. Example of terrible search below
<img width="1205" alt="image" src="https://user-images.githubusercontent.com/55838845/197189362-d74f4c4c-7b63-4c29-bf9e-9356e1e8f916.png">

OpenVerse however also sources images from wikipedia commons, which has a much better search capability 
I have updated the images to use image urls from wikipedia commons, using their no-restrictions filter. I am hoping that is ok for the publicly accessible URLs. They load fine in the editor and FE.

Thanks

faisal-alvi reviewed (changes requested):
@cldhmmr thank you for working on this. The changes look good, however, I have observed a few things. Please check the following details.

1. The cover block seems okay but when I switched to the storefront theme, the text is not clearly visible, should we add a color so it's white regardless of the selected theme?

![image](https://user-images.githubusercontent.com/25176325/198966917-e11542cf-d6a4-48e5-82aa-205f0d9064be.png)

2. The text "Monthly Unique Views on 10up.com" should be "Monthly Page Views on 10up.com"

![image](https://user-images.githubusercontent.com/25176325/198967787-cb202fb7-8307-4672-be54-f421d157ba9d.png)

3. 
> I haven't been able to replicate an issue with the table itself - the table show a horizontal overflow scrollbar on mobile and the user can slide that. Could you verify please?

Yes, there is a horizontal bar. I am seeing a bit of big fonts in the `storefront` theme. Also, I have seen the variations in the font size.

For example, 23px in the "Digital Ad Specs":

![image](https://user-images.githubusercontent.com/25176325/198971024-5e5039f3-a46f-41f3-8251-74d94cdefb0c.png)

And 16px in the "Our Rates":

![image](https://user-images.githubusercontent.com/25176325/198971073-996159b8-e690-4b12-9d04-36a4c1d7e63d.png)

However, if I check in the `twentytwentyone` theme, the font size is same in both sections.

![image](https://user-images.githubusercontent.com/25176325/198971874-2e19c040-1f64-4b6e-9f72-fdeb7a1ecaf7.png)

![image](https://user-images.githubusercontent.com/25176325/198972016-410b1461-de63-45f9-9509-1c5bd26e533c.png)

4. 
> I am a little bit hesitant about the images being outdated gender archetypes and would suggest we change to something non-gendered like animals or lego, or something else. Thoughts on this appreciated.

@jeffpaul any thoughts on ^this point?

5. Cypress tests failing.
> I don't understand why this is being thrown or how to fix this? I'm not sure if related to the updates or not. Please let me know if you have any ideas?

I have checked in WordPress 5.7, the block patterns are broken.

![image](https://user-images.githubusercontent.com/25176325/198979587-02c8a2c2-b9e4-435c-9590-5c40f34da7fc.png)

This is the reason why the Cypress test is failing. Seem like the markup is different in the 5.7 and the latest version. Can you please check? I think we might have to bump a WP version. cc: @jeffpaul 

6. Thank you for the details on the image search. Just wanted to know if image URLs from wikipedia commons will be accepted when we try to submit these block patterns to the WordPress pattern library.

jeffpaul commented:
@faisal-alvi sounds like it'll be best if you can help pick up those suggested updates to help move the PR along, please and thanks!

faisal-alvi committed:
color fix in storefront for cover block

faisal-alvi committed:
stats block text change

faisal-alvi committed:
fix: variations in the font size

faisal-alvi committed:
cover block support for wp 5.7

faisal-alvi committed:
fix audience profile patter for wp 5.8

faisal-alvi committed:
WP min required version bump to 5.8

faisal-alvi committed:
Merge branch 'develop' into feature/remove-css

faisal-alvi commented:
Note: Bumping the WP minimum required version from `5.7` to `5.8` because, after this PR, the block patterns will be broken in 5.7 or below.

faisal-alvi commented:
@jeffpaul can you please request someone from the OSP team for a code review + QA testing? I've tried requesting `10up/open-source-practice` but it didn't re-requested anyone.

faisal-alvi committed:
remove extra line from readme file

jeffpaul commented:
@10up/open-source-practice somethings amiss on GH code reviews, so hoping one of you can chime in with review here... thanks!

dkotter commented on assets/css/frontend/components/index.css:
```diff
@@ -1,9 +1 @@
 /* Components */
```
Any reason to keep this file anymore? Seems like we've gotten rid of all component files, wondering if we should just delete that directory all together?

dkotter commented:
I've done an initial code review and things look good. Haven't had time to actually test yet though. Just one question: How does this impact anyone that's already using these patterns? Just want to ensure we don't push out an update that breaks any existing content created with the old patterns.

peterwilsoncc commented:
> Just want to ensure we don't push out an update that breaks any existing content created with the old patterns.

To cover this situation, I suggest maintaining some JavaScript and using [the deprecation functionality of the block editor](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/). I think this is possible when migrating to different blocks but it is possibly quite fiddly.

faisal-alvi commented:
> Just want to ensure we don't push out an update that breaks any existing content created with the old patterns.

Our plugin has only one custom block, the "Tabs" block, which is used in the block patterns. Other than that, everything else that is used in the block patterns is the core blocks. We have not changed anything in our custom block (Tabs block), so this PR will not affect any existing content created with old patterns. I've also tested it again and nothing is found broken.

### The `develop` branch with old block patterns:

![image](https://user-images.githubusercontent.com/25176325/206976493-45df2aca-95db-4ee8-bd88-7fd6cd046755.png)

### This PR branch with the same old block patterns:

![image](https://user-images.githubusercontent.com/25176325/206976554-e03c4eda-afcb-4355-9795-784c01ba2ade.png)

jeffpaul commented:
@faisal-alvi let's discuss the goals of this PR in our next 1:1 and determine the best path forward

iamdharmesh commented:
For reference: 
- [Patterns visauls with `develop` branch](https://user-images.githubusercontent.com/10613171/210967481-ade84ac5-2fa0-4670-9814-edf6ec5f2ca9.png)
- [Pattens visauls with this branch](https://user-images.githubusercontent.com/10613171/210967505-b7cdea67-984b-45ab-8001-36fe6e21318c.png)

faisal-alvi commented:
Update: Jeff and I spoke about not merging this PR into the plugin and keeping all the CSS to maintain the original design. In order to submit them to the [WordPress block pattern directory](https://wordpress.org/patterns/), the primary goal of this PR was to design a block pattern without CSS (because WP does not accept custom CSS).

The next step will be to work on creating draft patterns directly in the WP directory and if that meets our expectations, we will submit them to the WordPress directory. A FEE will be assigned for this work.

_Screenshot of the editor in the WP directory:_
![image](https://user-images.githubusercontent.com/25176325/211775177-89699b32-f2a5-4fad-ad8c-c12d1e4501fe.png)

The minimum WordPress version bump to 5.8 was one modification that could be preserved because block patterns were broken in that version and below. However, after testing, I discovered that block patterns are not broken in the `develop` branch, so we may leave that change in place as well.

Closing this PR.
